scripts/specs.ts

```diff
@@ -1,18 +1,54 @@
+import assert from "node:assert/strict";
+
 import webSpecs from 'web-specs' assert { type: 'json' };
 
 import features from '../index.js';
 
-const specUrls = new Set(webSpecs.map(spec => spec.nightly.url));
+const specUrls = webSpecs.map(spec => new URL(spec.nightly.url));
+
+type exception = [url: string, message: string];
+const exceptions: exception[] = [
+    [
+        "https://wicg.github.io/navigation-api/",
+        "This spec is moving to WHATWG HTML. Remove this exception when https://github.com/whatwg/html/pull/8502 merges."
+    ],
+];
+
+function isOK(url: URL, allowlist: exception[] = exceptions) {
+    for (const specUrl of specUrls) {
+        if (specUrl.protocol !== url.protocol || specUrl.origin !== url.origin) {
+            continue;
+        }
+        if (url.pathname.startsWith(specUrl.pathname)) {
```

Comment:
This will also allow some plausible incorrect spec URLs to persist, like https://w3c.github.io/gamepad/extensions.html if it is folded into https://w3c.github.io/gamepad/. And some less plausible broken URLs like https://w3c.github.io/gamepad/extensions.htmltypo.

The problem is HTML multipage of course. I think we should have the rule that if `specUrl.pathname` ends with "/multipage/" then anything goes, and unfortunately we won't catch broken URLs there. For other URLs, the pathname must match exactly, but `url.hash` can be whatever.

Reply:
I think eb63176 does what you're describing. Want to take another look?

> if `specUrl.pathname` ends with "/multipage/" then anything goes, and unfortunately we won't catch broken URLs there

Correct. Previously, I considered confirming that linked URLs (that don't exactly match `specUrl`) and hashes actually exist, but I know from experience that sort of thing is too slow to run all the time. If we're worried about this, we could put in a more thorough check and have it run only sometimes (e.g., have it run on some schedule and open an issue when it fails, or run only on dependabot upgrades to `web-specs`).

Reply:
I've filed https://github.com/w3c/browser-specs/issues/931 about making it easier for us to check URLs.